tools/lsp/language.rs

```diff
@@ -97,24 +98,170 @@ pub fn request_state(ctx: &std::rc::Rc<Context>) {
     }
 }
 
+fn builtin_component_info(name: &str) -> ComponentInformation {
```

Comment:
It would probably be cleaner to put all this new code in a new module.  maybe `mod element_catalog` ?


(the build_import_statements_edits could also go in that module)